tests/infrastructure/test_kubernetes_job.py

```diff
@@ -1081,26 +1081,26 @@ def mock_stream(*args, **kwargs):
                 func=mock_k8s_batch_client.list_namespaced_job,
                 field_selector=mock.ANY,
                 namespace=mock.ANY,
-                timeout_seconds=pytest.approx(1, abs=0.2),
+                timeout_seconds=2,
```

Comment:
As with the other test, we want to actually show that time is subtracted on each call. We should sleep for an integer in the `mock_stream`

Reply:
Maybe we want to use a fixture like `mock_anyio_sleep` for the monotonic clock to keep the tests fast?